### PR TITLE
165 refactor state

### LINKED ISSUE #165: Refactor State
State class is too big, with too many methods.
We are going to split this data class!

- [x] DistanceCheck (to Track?)
- [x] Current skin to skin class
- [x] Track must become Singleton (resettable)
- [x] Savegame related (getdefaultsavedir() to Savegame)

### PR DESCRIPTION
**Pull request by MathiasMeuleman**
Will close #165 

Not that much has changed.
Most of the changes include moving around some fields and methods.
Only big difference is Track is now a Singleton.

Please review!

<review

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/gui/skin/Skin.java [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
+
+
 /**
  * @author Maikel on 08/10/2015.
  */
 @SuppressWarnings("restriction")
 public class Skin {
     /**
-     * IRON MAN skin for player.
+     * IRON MAN currentSkin for player.
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/gui/skin/Skin.java]: Some more `currentSkin`s in this class.

<review, documentation

File: src/main/java/nl/tudelft/ti2206/group9/level/State.java [hunk 2]
```diff
@@ -25,22 +19,13 @@
     /** Highest score ever obtained. */
     private static double highscore;
 
-    /** Skin to be used. */
-    private static Skin skin;
-
     /** Current track, contains all entities. */
-    private static Track track = new Track();
+    private static Track track;
```
Comment (mpsijm) [inline src/main/java/nl/tudelft/ti2206/group9/level/State.java]: Is this field still used in State? It should be removed as Track is a Singleton now.

<review, logic, interface

File: src/test/java/nl/tudelft/ti2206/group9/level/StateTest.java [hunk 3]
```diff
@@ -107,13 +108,13 @@ public void testSetPlayerName() {
 
     @Test
     public void testGetDefaultSaveDir() {
```
Comment (mpsijm) [inline src/test/java/nl/tudelft/ti2206/group9/level/StateTest.java]: This test should not be part of StateTest anymore, just like the Track distance tests.

<review, testing

**Comment by mdingjan** [comment]
I reviewed your changes and fixed the last Checkstyle violation for you (which was about a restriction suppresswarnings, and I added comments for this). Your branch is now SAT violation free.

Please have a look at the comments above, merge the master into your branch and try to implement a EquippedItem class, which we already discussed. :)

After that, this branch is ready for merging!

<review

**Comment by MathiasMeuleman** [comment]
Okey, I applied all comments in this PR. Also, there is now a new CurrentItems class in the shop package that stores the current skin and current soundtrack. Please take another look!

<review

**Comment by thervh70** [comment]
I approve these changes. State is really trimmed down neatly.
I was shocked by the size of the diff, but it is turned out State really was too big.
Ready for merge!

< review

**Comment by mpsijm** [comment]
Thanks Mitchell, for processing my feedback for Mathias. As you had nothing more to comment as well, I'll merge this once Travis is happy :)

<review

**Comment by mdingjan** [comment]
I reviewed your changes. They are looking good.
I agree with the points described above, therefore I processed the feedback for you.

Ready for merging in my opinion.

<review

**Pull request merged**